python/src/aac/plugins/validators/_validator_result.py

```diff
@@ -0,0 +1,18 @@
+from attr import attrib, attrs, validators, Factory
+
+
+@attrs(slots=True, auto_attribs=True)
+class ValidatorResult:
+    """Represents the validation result from a validator plugin on a single definition.
+
+    Attributes:
+        messages (list): A dictionary of definition names as keys to lists of messages
+        is_valid (bool): A bool indicating whether the definition is valid or not (True/False)
+    """
+
+    messages: list[str] = attrib(default=Factory(list), validator=validators.instance_of(list))
+    is_valid: bool = attrib(default=False, validator=validators.instance_of(bool))
+
+    def get_messages_as_string(self) -> str:
```

Comment:
Hmm. I'll think on this and circle back.

Reply:
I updated the plugin execution result to match, but I don't think it's worth forcing the classes to re-use some functions right now since the validator result will need to be adjusted for use with the LSP diagnostics feature.